pkg/apis/resource/validation/validation.go

```diff
@@ -825,6 +828,10 @@ var (
 		// optional dot-separated build identifier segments (e.g. +build.id.20240305)
 		`(\+` + buildIdentifier + `(\.` + buildIdentifier + `)*)?` +
 		`$`)
+
+	conditionTypeRegex = regexp.MustCompile(
```

Comment:
We should be able to reuse existing validation for a condition type...

https://github.com/kubernetes/kubernetes/blob/fcf22b3ebe00bdf5fb96691d9e05a0c01a661f83/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/validation/validation.go#L319

So below, just call `metav1validation.ValidateLabelName`.

Reply:
Thanks, I will use it.

Reply:
Updated to use validation present in metav1Validation.

Reply:
The fix for resourceSlice test was not included in the commit. I will add it.